docs/source/configuration.rst

```diff
@@ -0,0 +1,68 @@
+Configuration
+=============
+
+Experiments with NeuralMonkey are configured using configuration files
+which specifies the architecture of the model, meta-parameters of the
+learning, the data, the way the data are processed and the way the model
+is run.
+
+Syntax
+------
+
+The configuration files extends syntax of INI files. INI files contain
+*key-value pairs*, delimited by an equal sign (``=``) with no spaces
+around. Traditionally, everything on the right side of the equal sign is
+interpreted as a string. The key-value pairs can be grouped into
+*sections*. Section starts with its header which consists of the section
+name in square brackets. Everything below the header is considered a
+part of the section. The files can also contain comments starting with
+semi-colons (``;``). You can find more details on INI syntax, e.g., on
+the corresponding `Wikipedia
+page <https://en.wikipedia.org/wiki/INI_file>`__.
+
+The configuration introduces several additional constructs for the
+values. There are both atomic values, and compound values.
+
+The atomic values are:
+
+-  booleans: literals ``True`` and ``False``
+
+-  integers: strings that could be interpreted as integers by Python
+   (e.g., ``1``, ``002``)
+
+-  floats: string that could be interpreted as floats by Python (e.g.,
+   ``1.0``, ``.123``, ``2.``, ``2.34e-12``)
+
+-  strings: string literals in quotes (e.g., ``"walrus"``, ``"5"``)
+
+-  Python names: string literals without quotes which are neither
```

Comment:
Tady je vůbec divný psát `string literals`, psal bych `string(s)` stejně jako u floatů a intergerů (kde by se mělo sjednotit gramatické číslo toho slova). A především je to špatně, protože do toho spadnou i `section references`, které jsou definované v dalším bodě a měly by být vyjmenované v tom seznamu věcí, které nejsou `Python names`.